gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java

```diff
@@ -94,7 +96,14 @@ public LocalFsSpecProducer(Config config) {
    * @param headers*/
   @Override
   public Future<?> deleteSpec(URI deletedSpecURI, Properties headers) {
-    return new CompletedFuture<>(Boolean.TRUE, null);
+    String[] uriTokens = deletedSpecURI.getPath().split("/");
+    String jobFileName = String.join("_", uriTokens) + ".job";
+    File file = new File(jobFileName);
+    if (file.delete()) {
+      log.info("Deleted spec: {}", jobFileName);
+      return new CompletedFuture<>(Boolean.TRUE, null);
+    }
+    throw new RuntimeException(String.format("Failed to delete file with uri %s", deletedSpecURI));
```

Comment:
This shouldn't halt the service, return a `CompletedFuture` with an exception and expect the invoker to do proper error handling.

Reply:
That's a good point actually. In the other `SpecProducers` they throw a runtime exception as well, would it affect anything?